src/NewRelic.Telemetry/Metrics/NewRelicMetric.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+
+namespace NewRelic.Telemetry.Metrics
+{
+    public readonly struct NewRelicMetric
+    {
+        public string Type { get; }
+
+        public Dictionary<string, object>? Attributes { get; }
+
+        public long? Timestamp { get; }
+
+        public string Name { get; }
+
+        [DataMember(Name = "interval.ms")]
+        public long? IntervalMs { get; }
+
+        [DataMember(Name = "value")]
+        public object? ValueForSerialization => SummaryValue ?? Value as object;
+
+        [IgnoreDataMember]
+        public double? Value { get; }
+
+        [IgnoreDataMember]
+        public NewRelicMetricSummaryValue? SummaryValue { get; }
+
+        private NewRelicMetric(string type, long? timestamp, string name, Dictionary<string, object>? attributes, long? intervalMs, double? value, NewRelicMetricSummaryValue? summaryValue)
+        {
+            Type = type;
+            Timestamp = timestamp;
+            Name = name;
+            Attributes = attributes;
+            IntervalMs = intervalMs;
+            Value = value;
+            SummaryValue = summaryValue;
+        }
+        
+        public static NewRelicMetric CreateCountMetric(string name, long? timestamp, Dictionary<string, object>? attributes, double value, long intervalMs)
+        {
+            return new NewRelicMetric("count", timestamp, name, attributes, intervalMs, value, null);
+        }
+
+        public static NewRelicMetric CreateGaugeMetric(string name, long? timestamp, Dictionary<string, object>? attributes, double value)
```

Comment:
Todo:  In the prior version of a Gauge metric, it seemed to support an interval.  But the spec does not indicate that it should.